:farmer: Rci__nightly-fastrtps/connext test_parameter_event_handler failing RCLerror

### Generated by Generative AI

_No response_

### Operating System:

Ubuntu 24.04

### ROS version or commit hash:

Rolling

### RMW implementation (if applicable):

rmw_connext, rmw_cyclonedds

### RMW Configuration (if applicable):

_No response_

### Client library (if applicable):

rclpy and rclcpp

### 'ros2 doctor --report' output

_No response_

### Steps to reproduce issue

1. Go to any of the jobs referenced below
2. Build the job
3. See tests failing

### Expected behavior

To build without test regressions

### Actual behavior

Building with tests regresisons

### Additional information

### Reference Builds
- [Rci__nightly-cyclonedds_ubuntu_noble_amd64#678](http://build.ros2.org/job/Rci__nightly-cyclonedds_ubuntu_noble_amd64/678) with rclpy
- [Rci__nightly-connext_ubuntu_noble_amd64#685](http://build.ros2.org/job/Rci__nightly-connext_ubuntu_noble_amd64/685) with rclpy
- [Rci__nightly-connext_ubuntu_noble_amd64#685](http://build.ros2.org/job/Rci__nightly-connext_ubuntu_noble_amd64/685) with rclcpp

### Failing tests
- projectroot.test_parameter_event_handler
- rclpy.rclpy.test.test_parameter_event_handler.TestParameterEventHandler.test_configure_nodes_filter_with_check_add_parameter_callback
- rclpy.rclpy.test.test_parameter_event_handler.TestParameterEventHandler.test_configure_nodes_filter_with_check_add_parameter_event_callback
- projectroot.test.rclcpp.test_parameter_event_handler
- rclcpp.TestNode.ConfigureNodesFilterAndCheckAddParameterEventCallback
- rclcpp.TestNode.ConfigureNodesFilterAndCheckAddParameterCallback

### Log Output
```
        def callback_remote_node2(param: ParameterMsg) -> None:
            nonlocal received_event_from_remote_node2
            if param.name == remote_node2_param_name:
                received_event_from_remote_node2 = True
    
        # Configure to only receive parameter events from remote_node_name2
>       assert self.parameter_event_handler.configure_nodes_filter(
            [f'/rclpy/{remote_node_name2}'])

../../src/ros2/rclpy/rclpy/test/test_parameter_event_handler.py:413: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../src/ros2/rclpy/rclpy/rclpy/parameter_event_handler.py:400: in configure_nodes_filter
    self.parameter_event_subscription.set_content_filter(filter_expression, quoted_node_names)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <rclpy.subscription.Subscription object at 0x79b48b99a9c0>
filter_expression = 'node = %0'
expression_parameters = ["'/rclpy/remote_node_2'"]

    def set_content_filter(self, filter_expression: str, expression_parameters: list[str]) -> None:
        """
        Set the filter expression and expression parameters for the subscription.
    
        :param filter_expression: The filter expression to set.
        :param expression_parameters: The expression parameters to set.
        :raises: RCLError if internal error occurred when calling the rcl function.
        """
        with self.handle:
>           self.__subscription.set_content_filter(filter_expression, expression_parameters)
E           test_rclpy._rclpy_pybind11.RCLError: Failed to set cft expression parameters: unimplemented, at /tmp/ws/src/ros2/rmw_fastrtps/rmw_fastrtps_dynamic_cpp/src/rmw_subscription.cpp:174, at /tmp/ws/src/ros2/rcl/rcl/src/rcl/subscription.c:502

../../src/ros2/rclpy/rclpy/rclpy/subscription.py:222: RCLError
```

### Flakyness 15 days
job_name|last_fail|first_fail|build_count|failure_count|failure_percentage
|--|--|--|--|--|--|
Rci__nightly-cyclonedds_ubuntu_noble_amd64|2026-02-13|2026-02-05|6|6|100.0
Rci__nightly-connext_ubuntu_noble_amd64|2026-02-12|2026-02-05|5|5|100.0
Rci__nightly-fastrtps-dynamic_ubuntu_noble_amd64|2026-02-11|2026-02-05|3|3|100.0

CC: @Crola1702

--- a/rclpy/test/test_parameter_event_handler.py
+++ b/rclpy/test/test_parameter_event_handler.py
@@ -301,6 +301,10 @@ class TestParameterEventHandler(unittest.TestCase):
         assert '/test_node' == self.parameter_event_handler._resolve_path('/test_node')
 
     def test_configure_nodes_filter_with_check_add_parameter_event_callback(self) -> None:
+        if rclpy.get_rmw_implementation_identifier() != 'rmw_fastrtps_cpp' and \
+           rclpy.get_rmw_implementation_identifier() != 'rmw_connextdds':
+            pytest.skip('Content filter is now only supported in FastDDS and ConnextDDS.')
+
         remote_node_name1 = 'remote_node_1'
         remote_node1 = rclpy.create_node(
             remote_node_name1,
@@ -380,6 +384,10 @@ class TestParameterEventHandler(unittest.TestCase):
         remote_node2.destroy_node()
 
     def test_configure_nodes_filter_with_check_add_parameter_callback(self) -> None:
+        if rclpy.get_rmw_implementation_identifier() != 'rmw_fastrtps_cpp' and \
+           rclpy.get_rmw_implementation_identifier() != 'rmw_connextdds':
+            pytest.skip('Content filter is now only supported in FastDDS and ConnextDDS.')
+
         remote_node_name1 = 'remote_node_1'
         remote_node1 = rclpy.create_node(
             remote_node_name1,

